arch/xtensa/src/esp32s2/esp32s2_tim_lowerhalf.c

```diff
@@ -0,0 +1,595 @@
+ ****************************************************************************/
+
+static int esp32s2_timer_maxtimeout(FAR struct timer_lowerhalf_s *lower,
+                                  uint32_t *max_timeout)
+{
+  DEBUGASSERT(max_timeout);
+
+  *max_timeout = UINT32_MAX;
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: esp32s2_setcallback
+ *
+ * Description:
+ *   Set the provided callback to be called at timeout from withing the
+ *   ISR.
+ *
+ * Input Parameters:
+ *   lower    - A pointer to the publicly visible representation of
+ *              the "lower-half" driver state structure.
+ *   callback - A pointer to the callback. If this function pointer
+ *              is NULL, then the reset-on-expiration behavior is restored.
+ *   arg      - A pointer to the argument that will be provided to
+ *              the callback
+ *
+ ****************************************************************************/
+
+static void esp32s2_timer_setcallback(FAR struct timer_lowerhalf_s *lower,
+                                      tccb_t callback, FAR void *arg)
+{
+  FAR struct esp32s2_timer_lowerhalf_s *priv =
+    (FAR struct esp32s2_timer_lowerhalf_s *)lower;
+  irqstate_t flags;
+  int ret = OK;
+
+  DEBUGASSERT(priv);
+
+  /* Save the new callback */
+
+  priv->callback = callback;
+  priv->arg      = arg;
+
+  flags = enter_critical_section();
+
+  /* There is a user callback and the timer has already been started */
+
+  if (callback != NULL && priv->started == true)
+    {
+      ret = ESP32S2_TIM_SETISR(priv->tim, esp32s2_timer_handler, priv);
+      ESP32S2_TIM_ENABLEINT(priv->tim);
+    }
+  else
+    {
+      ESP32S2_TIM_DISABLEINT(priv->tim);
+      ret = ESP32S2_TIM_SETISR(priv->tim, NULL, NULL);
+    }
+
+  leave_critical_section(flags);
+  if (ret != OK)
+    {
+      tmrerr("Error to set ISR: %d", ret);
+    }
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: esp32s2_timer_initialize
+ *
+ * Description:
+ *   Bind the configuration timer to a timer lower half instance and
+ *   register the timer drivers at 'devpath'
+ *
+ * Input Parameters:
+ *   devpath - The full path to the timer device.  This should be of the
+ *             form /dev/timerx.
+ *   timer - the timer's number.
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; A negated errno value is returned
+ *   to indicate the nature of any failure.
+ *
+ ****************************************************************************/
+
+int esp32s2_timer_initialize(FAR const char *devpath, uint8_t timer)
+{
+  struct esp32s2_timer_lowerhalf_s *lower = NULL;
+  int ret = OK;
+
+  DEBUGASSERT(devpath);
+
+  switch (timer)
+    {
+#ifdef CONFIG_ESP32S2_TIMER0
+      case 0:
```

Comment:
If those `TIMERx` defines are kept on the Timer interface, they could be used here for indexing the available timers.